### PR TITLE
Email verfication

### PR DESCRIPTION
**Pull request by Saphereye**
## Description
Added middleware to only allow bitsian to access the docs.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New website feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] New script (new python script for handouts handling)
- [ ] New resource (new file that is added)
  - [ ] Adding new / updating notes
  - [ ] Adding new / updating handouts
  - [ ] Adding new / updating chronicles
  - [ ] Adding new / updating reviews
- [ ] Misc (explain in description)

## Checklist:

- [X] I used `pnpm` for my project dependencies
- [X] I didnt push any `.env` files

### CODE REVIEW

File: middleware.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { NextResponse } from 'next/server';
+import type { NextRequest } from 'next/server';
+import { getServerSession } from 'next-auth/next';
+import { authOptions } from './pages/api/auth/[...nextauth]';
+
+const EMAIL_REGEX = /^(?:[fh]\d{8}@(hyderabad|pilani)\.bits-pilani\.ac\.in|[fh]\d{8}[pgh]@alumni\.bits-pilani\.ac\.in)$/;
+// const EMAIL_REGEX = /a/;
```
Comment (thenicekat) [inline middleware.ts:7]: remove comment.

File: middleware.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      { status: 401 }
+    );
+  }
+
+  const email = session?.user?.email;
+
+  if (!email || !EMAIL_REGEX.test(email)) {
+    return NextResponse.json(
+      {
+        message: 'Access restricted to BITS Hyderabad or Pilani students/alumni',
```
Comment (thenicekat) [inline middleware.ts:28]: what did goa and dubai do ;-;

Comment (Saphereye) [reply middleware.ts:28]: fake campuses smh (I will test the regex once properly)

Comment (Saphereye) [reply middleware.ts:28]: I captured all I remember: https://regex101.com/r/EX9llp/1

Comment (thenicekat) [reply middleware.ts:28]: Lite

**Comment by Saphereye** [comment]
> Remove extra comment. LGTM.

Pushed

**Comment by thenicekat** [comment]
Reverted this PR due to build fail issues

**Pull request merged**